src/WaveTrack.cpp

```diff
@@ -696,27 +851,27 @@ Track::Holder WaveTrack::Copy(double t0, double t1, bool forClipboard) const
          WaveClip *const newClip = newTrack->mClips.back().get();
          newClip->Offset(-t0);
       }
-      else if (t1 > clip->GetStartTime() && t0 < clip->GetEndTime())
+      else if (t1 > clip->GetPlayStartTime() && t0 < clip->GetPlayEndTime())
       {
          // Clip is affected by command
          //wxPrintf("copy: clip %i is affected by command\n", (int)clip);
 
-         const double clip_t0 = std::max(t0, clip->GetStartTime());
-         const double clip_t1 = std::min(t1, clip->GetEndTime());
+         const double clip_t0 = std::max(t0, clip->GetPlayStartTime());
+         const double clip_t1 = std::min(t1, clip->GetPlayEndTime());
 
          auto newClip = std::make_unique<WaveClip>
             (*clip, mpFactory, ! forClipboard, clip_t0, clip_t1);
          newClip->SetName( 
             //i18n-hint Template for clip name generation used when copying a part of clip data
             XC("%s samples [%.2f-%.2f]", "clip name template")
-                 .Format(clip->GetName(), clip_t0 - clip->GetStartTime(), clip_t1 - clip->GetStartTime())
+                 .Format(clip->GetName(), clip_t0 - clip->GetPlayStartTime(), clip_t1 - clip->GetPlayStartTime())
                  .Translation());
 
          //wxPrintf("copy: clip_t0=%f, clip_t1=%f\n", clip_t0, clip_t1);
 
          newClip->Offset(-t0);
-         if (newClip->GetOffset() < 0)
-            newClip->SetOffset(0);
+         if (newClip->GetPlayStartTime() < 0)
+            newClip->SetPlayStartTime(0);
```

Comment:
A definite mistake here.  WaveClip::SetOffset used to affect the envelope and the sequence.  But WaveClip::SetPlayStartTime now affects only the sequence.